core/src/main/java/org/p2p/core/pricecache/PriceCacheRepository.kt

```diff
@@ -0,0 +1,23 @@
+package org.p2p.core.pricecache
+
+import java.math.BigDecimal
+import org.p2p.core.model.TokenPriceWithMark
+
+abstract class PriceCacheRepository {
```

Comment:
why not composition?
you can create TokenPricesInMemoryRepository and just inject that
inheritance is evil >:(